.circleci/config.yml

```diff
@@ -388,11 +388,26 @@ jobs:
       - slack/status:
           fail_only: true
           failure_message: ':aws-emoji: :red_circle: AWS Pinpoint country configuration is out of date'
+  check_changelog:
+    executor: ruby_browsers
+    steps:
+      - checkout
+      - run:
+          name: Check Changelog
+          command: |-
+            if [ -z "${CIRCLE_PULL_REQUEST}" ]
+            then
+              exit 0
+            else
+              ./scripts/changelog-check -b "${CIRCLE_BRANCH}" -s main
+            fi
+
 workflows:
   version: 2
   release:
     jobs:
       - setup
+      - check_changelog
```

Comment:
quick thought, is there a way to say "only run this when making a PR that targets `main`"? Otherwise RC branches will run checking against `main` but the target will be `stages/prod` right?

Reply:
that's WIP, yeah. Unfortunately CircleCI doesn't include the target of the PR out of the box (https://discuss.circleci.com/t/create-a-circle-target-branch-envar/10022/4)

Reply:
Unfortunately this doesn't seem resolvable without a GitHub token in CircleCI which I'd rather not figure out right now 😛

I've added a condition that should allow most of our RC branches to bypass the check in https://github.com/18F/identity-idp/pull/5836/commits/4d769b6fe9070c3a4fa626c5c543ea4ebe3cf38f. It also _should_ be ok for the most part since the diff of an RC into `stages/prod` should include changelog additions.

Reply:
On some basic testing, it doesn't seem like the token is enforced by the API endpoint? Following [this guide](https://discuss.circleci.com/t/how-to-retrieve-a-pull-requests-base-branch-name-github/36911), I was able to get the base branch by:

```
curl https://api.github.com/repos/${CIRCLE_PULL_REQUEST:19} | jq .base.ref
```

Reply:
Thanks for looking into that, good to know there is an unauthenticated option. Is it possible to create an org or team token?

Reply:
Another option is we do a one-off GitHub Action for this one task, which includes branch filtering out of the box:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters